src/components/views/messages/MPollBody.tsx

```diff
@@ -43,74 +44,93 @@ import PollCreateDialog from "../elements/PollCreateDialog";
-        ),
-    ]);
+    const relationsList: Relations[] = [];
+
+    const pollResponseRelations = getRelationsForEvent(
+        eventId,
+        "m.reference",
+        M_POLL_RESPONSE.name,
+    );
+    if (pollResponseRelations) {
+        relationsList.push(pollResponseRelations);
+    }
+
+    const pollResposnseAltRelations = getRelationsForEvent(
+        eventId,
+        "m.reference",
+        M_POLL_RESPONSE.altName,
+    );
+    if (pollResposnseAltRelations) {
+        relationsList.push(pollResposnseAltRelations);
+    }
+
+    return new RelatedRelations(relationsList);
 }
 
 export function findTopAnswer(
     pollEvent: MatrixEvent,
     matrixClient: MatrixClient,
-    getRelationsForEvent?: (
-        eventId: string,
-        relationType: string,
-        eventType: string
-    ) => Relations,
+    getRelationsForEvent?: GetRelationsForEvent,
 ): string {
     if (!getRelationsForEvent) {
         return "";
     }
 
+    const pollEventId = pollEvent.getId();
+    if (!pollEventId) {
+        logger.warn(
+            "findTopAnswer: Poll event needs an event ID to fetch relations in order to determine " +
+            "the top answer - assuming no best answer",
+        );
+        return "";
+    }
+
     const poll = pollEvent.unstableExtensibleEvent as PollStartEvent;
     if (!poll?.isEquivalentTo(M_POLL_START)) {
-        console.warn("Failed to parse poll to determine top answer - assuming no best answer");
+        logger.warn("Failed to parse poll to determine top answer - assuming no best answer");
         return "";
     }
 
     const findAnswerText = (answerId: string) => {
         return poll.answers.find(a => a.id === answerId)?.text ?? "";
     };
 
-    const voteRelations = createVoteRelations(getRelationsForEvent, pollEvent.getId());
-
-    const endRelations = new RelatedRelations([
-        getRelationsForEvent(
-            pollEvent.getId(),
-            "m.reference",
-            M_POLL_END.name,
-        ),
-        getRelationsForEvent(
-            pollEvent.getId(),
-            "m.reference",
-            M_POLL_END.altName,
-        ),
-    ]);
+    const voteRelations = createVoteRelations(getRelationsForEvent, pollEventId);
+
+    const relationsList: Relations[] = [];
+
+    const pollEndRelations = getRelationsForEvent(
+        pollEventId,
+        "m.reference",
+        M_POLL_END.name,
+    );
+    if (pollEndRelations) {
+        relationsList.push(pollEndRelations);
+    }
+
+    const pollEndAltRelations = getRelationsForEvent(
+        pollEventId,
+        "m.reference",
+        M_POLL_END.altName,
+    );
+    if (pollEndAltRelations) {
+        relationsList.push(pollEndAltRelations);
+    }
+
+    const endRelations = new RelatedRelations(relationsList);
 
     const userVotes: Map<string, UserVote> = collectUserVotes(
         allVotes(pollEvent, matrixClient, voteRelations, endRelations),
-        matrixClient.getUserId(),
+        null,
```

Comment:
All params are now optional. It looks imho less weird if we would remove them instead of calling with `null`.